internal/queue/storage_queue.go

```diff
@@ -0,0 +1,116 @@
+package queue
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/DIMO-Network/edge-network/internal/constants"
+	"github.com/google/uuid"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+)
+
+type StorageQueue interface {
+	Enqueue(message string) error
+	Dequeue() ([]Message, error)
+}
+
+type storageQueue struct {
+	mu     sync.Mutex
+	unitID uuid.UUID
+}
+
+type Message struct {
+	Time    time.Time `json:"time"`
```

Comment:
I wonder if we make this struct more specific to our use case: 
- signal_name
- value
- timestamp